airflow/models/dag.py

```diff
@@ -2880,12 +2883,16 @@ def dags_needing_dagruns(cls, session: Session):
         # TODO[HA]: Bake this query, it is run _A lot_
         # We limit so that _one_ scheduler doesn't try to do all the creation
         # of dag runs
+        # exclude dags with import error
+        import_errors = [c.filename for c in session.query(importerror.filename).all()]
         query = (
             session.query(cls)
             .filter(
+                cls.max_active_runs != None,  # noqa
                 cls.is_paused == expression.false(),
                 cls.is_active == expression.true(),
                 cls.next_dagrun_create_after <= func.now(),
+                cls.fileloc.notin_(import_errors),
```

Comment:
We should find another way of doing this -- this function is part of the "hot" path of the scheduler, and a) adding an extra query and by sending (possibly) a large number of params via this "NOT IN (?,?,?)" etc is not a great idea.

Reply:
I have updated it to exclude those with null max_active_runs.